### PR TITLE
Set up GitGaurdian pre-commit hook

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Adds ggshield pre-commit hook

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [x] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
- Update your local branch with `git pull origin` and switch to this branch with `git checkout --track origin/ft-secrets-security`.
- Follow the ggshield pre-commit [setup tutorial](https://docs.google.com/presentation/d/1vIoC-g9UmcAVarjEQgXpBK_bVbACaXLn3VzY8AWg6hU/edit#slide=id.g11b1ae9a83c_0_27) and attempt to commit a secret in your local `ft-secrets-security` branch. The fail message should list the file and/or line in which the secret you attempted to commit is. Share a screenshot of that fail message in a comment on this PR.
- Be sure to approve this PR if all went well. It helps determine who was successful and who might need some assistance. We shall merge the PR when all reviewers were successful and approved.

#### What are the relevant tickets?
- [PLAT-1078](https://airqoteam.atlassian.net/browse/PLAT-1078)

#### Screenshots (optional)

### CODE REVIEW

File: .gitsecret/paths/mapping.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.env:1287489e9d838c379d3bdff223da342eab1ebdcc182dbac2e4c9cf723b84cf9d
```
Comment (NoahNsimbe) [inline .gitsecret/paths/mapping.cfg:1]: Hi @123MwanjeMike are these supposed to be committed?

**Review by NoahNsimbe — COMMENTED** [review]
I see two files committed. `.env.secret` and `secret.secret`. My understanding is that a secret is  not supposed to be shared.

File: .gitsecret/paths/mapping.cfg (hunk 1 above)
Comment (123MwanjeMike) [reply .gitsecret/paths/mapping.cfg:1]: Yes @NoahNsimbe.
These are hashes only people with whom the secret have been shared are able to use to decrypt the file.
No worries.

**Comment by 123MwanjeMike** [comment]
> I see two files committed. `.env.secret` and `secret.secret`. My understanding is that a secret is not supposed to be shared.

You are right @NoahNsimbe. Secrets are not supposed to be shared to the outside world and that is the essence of git secret. It  enables us sync and share this secrets among team members whist keeping them private to the outside world.

**Comment by uman95** [comment]
LGTM

image

**Comment by 123MwanjeMike** [comment]
Hi all, please find the tutorial slides [here](https://docs.google.com/presentation/d/1vIoC-g9UmcAVarjEQgXpBK_bVbACaXLn3VzY8AWg6hU/edit#slide=id.gf3cbffa5d9_0_7)

**Comment by 123MwanjeMike** [comment]
> LGTM

Thanks @NoahNsimbe. Kindly attach the screenshot as well.

**Review by sserurich — APPROVED** [review]
Set it up and gitguardian was able to prevent secret from being commited
image

**Review by danielogen — APPROVED** [review]
Working well
Screen Shot 2022-05-26 at 9 33 10 AM

**Review by ja-odur — APPROVED** [review]
LGTM
Screenshot 2022-06-11 at 10 13 20

**Review by Mnoble-19 — APPROVED** [review]
Screenshot 2022-07-12 at 15 55 42

**Comment by Baalmart** [comment]
Hi @123MwanjeMike , thanks for this. I will be waiting for my followup session to get my issue resolved. Thanks.

**Comment by 123MwanjeMike** [comment]
> Hi @123MwanjeMike , thanks for this. I will be waiting for my followup session to get my issue resolved. Thanks.

Hi @Baalmart 
I think we can merge this PR since majority were successful in setting up.
We can follow up individually in the background for those that were not successful.

**Pull request merged**